### PR TITLE
EUI-5285 Hearing details page display api data

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-5285

### Change description ###
Hearings actuals-Retrieve hearing details for display in Hearing details page from 'Get /hearingActuals' API

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html [hunk 1]
```diff
@@ -1,4 +1,147 @@
...
+      <h1 class="govuk-heading-m">Participants</h1>
+      <span class="govuk-body align-right">
+        <a href="javascript:void(0)" [routerLink]="['/', 'hearing-actual', 'actuals-parties']" class="govuk-link">Update</a>
+      </span>
+      <div *ngFor="let actualHearingDay of hearingActualsMainModel.hearingActuals.actualHearingDays">
+        <dl class="govuk-summary-list">
+          <div class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">
+              {{actualHearingDay.actualDayParties[0].actualIndividualDetails.firstName}} {{actualHearingDay.actualDayParties[0].actualIndividualDetails.lastName}}
+              <br>
```
Comment (rayliangatos) [inline src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html:79]: if this actualHearingDay.actualDayParties[0] can be put into the typescript variable as this is used loads of time in the template?

File: src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html [hunk 2]
```diff
@@ -1,4 +1,147 @@
...
+<a class="govuk-back-link" href="javascript:history.back()" (click)="onBack()">Back</a>
+<main class="govuk-main-wrapper govuk-main-wrapper--auto-spacing" id="main-content" role="main">
+  <div class="govuk-grid-row" *ngIf="hearingActualsMainModel">
+    <div class="govuk-grid-column-two-thirds">
+      <span class="govuk-caption-l"></span>
+      <h1 class="govuk-heading-l">Hearing details</h1>
+      <dl class="govuk-summary-list">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Hearing date</dt>
+          <dd class="govuk-summary-list__value">{{hearingActualsMainModel.hearingActuals.actualHearingDays[0].hearingDate | date:"dd MMMM yyyy"}}</dd>
```
Comment (rayliangatos) [inline src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html:10]: could this hearingActualsMainModel.hearingActuals.actualHearingDays[0] put into the variable in ts as it's used many times in the template

File: src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html [hunk 3]
```diff
@@ -1,4 +1,147 @@
...
+      </dl>
+      <br>
+      <h1 class="govuk-heading-m">Hearing stage and result</h1>
+      <span class="govuk-body align-right">
+        <a href="javascript:void(0)" [routerLink]="['/', 'hearing-actual', 'stage-and-result']" class="govuk-link">Update</a>
+      </span>
+      <dl class="govuk-summary-list">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Hearing stage</dt>
+          <dd class="govuk-summary-list__value">{{hearingActualsMainModel.hearingActuals.hearingOutcome.hearingType}}</dd>
```
Comment (rayliangatos) [inline src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html:21]: if this hearingActualsMainModel.hearingActuals.hearingOutcome can put in the varialbe in tpyescript to get this shorter in the tempalte.

File: src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html:79]: Good suggestion @rayliangatos, I have now made the changes accordingly.

File: src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.html]: I have now updated the code.

File: src/hearings/models/hearingActualsMainModel.ts [hunk 4]
```diff
@@ -72,12 +73,11 @@ export interface ActualOrganisationDetailsModel {
...
 export interface ActualDayPartyModel {
-  actualPartyId: number;
+  actualPartyId: string;
   didNotAttendFlag: boolean;
   actualIndividualDetails: ActualIndividualDetailModel;
   actualOrganisationDetails: ActualOrganisationDetailsModel;
   partyChannelSubType: string;
-  partyId: number;
-  partyRole: number;
+  partyRole: string;
```
[the remark this answers is not in the document]
Comment (johnbenjamin-hmcts) [reply src/hearings/models/hearingActualsMainModel.ts:81]: In the API, ActualDayParty model contains the actialPartyId but not partyId.

**Review by rayliangatos — APPROVED** [review]
thanks for addressing the comments.

File: api/hearings/data/hearing-actuals.mock.data.ts [hunk 5]
```diff
@@ -192,8 +190,8 @@ export const HEARING_ACTUAL: HearingActualsMainModel = {
               cftOrganisationID: '54321',
               name: 'Company C',
             },
-            partyId: 4,
-            partyRole: 4,
+            partyId: '4',
```
Comment (SivaprakasamSivaprakasam) [inline api/hearings/data/hearing-actuals.mock.data.ts:193]: are we sure this partyid is a string, can't be defined as int?

File: src/hearings/models/hearingActualsMainModel.ts [hunk 6]
```diff
@@ -72,12 +73,11 @@ export interface ActualOrganisationDetailsModel {
 }
 
 export interface ActualDayPartyModel {
-  actualPartyId: number;
+  actualPartyId: string;
```
Comment (SivaprakasamSivaprakasam) [inline src/hearings/models/hearingActualsMainModel.ts:76]: are we sure this actualPartyId is a string, can't be defined as int?

**Review by SivaprakasamSivaprakasam — APPROVED** [review]
approved with few minor comments

**Pull request merged**